src/Symfony/Component/Cache/Adapter/MemcachedAdapter.php

```diff
@@ -0,0 +1,170 @@
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * @author Rob Frawley 2nd <rmf@src.run>
+ */
+class MemcachedAdapter extends AbstractMemcacheAdapter
+{
+    /**
+     * Construct adapter by passing a \Memcached instance and an optional namespace and default cache entry ttl.
+     *
+     * @param \Memcached  $client
+     * @param string|null $namespace
+     * @param int         $defaultLifetime
+     */
+    public function __construct(\Memcached $client, $namespace = '', $defaultLifetime = 0)
+    {
+        parent::__construct($namespace, $defaultLifetime);
+        $this->client = $client;
+    }
+
+    /**
+     * Factory creation method that provides an instance of this adapter with a\Memcached client instantiated and setup.
+     *
+     * Valid DSN values include the following:
+     *  - memcached://localhost                  : Specifies only the host (defaults used for port and weight)
+     *  - memcached://example.com:1234           : Specifies host and port (defaults weight)
+     *  - memcached://example.com:1234?weight=50 : Specifies host, port, and weight (no defaults used)
+     *
+     * Valid options include any client constants, as described in the PHP manual:
+     *  - http://php.net/manual/en/memcached.constants.php
+     *
+     * Options are expected to be passed as an associative array with indexes of the option type with coorosponding
+     * values as the option assignment.
+     *
+     * @param string|null $dsn
+     * @param array       $opts
+     * @param string|null $persistentId
+     *
+     * @return MemcachedAdapter
+     */
+    public static function create($dsn = null, array $opts = array(), $persistentId = null)
+    {
+        if (!extension_loaded('memcached')) {
+            throw new InvalidArgumentException('Failed to create Memcache client due to missing "memcached" extension.');
+        }
+
+        $adapter = new static(new \Memcached($persistentId));
+        $adapter->setup($dsn ? array($dsn) : array(), $opts);
+
+        return $adapter;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doSave(array $values, $lifetime)
+    {
+        return $this->client->setMulti($values, $lifetime)
+            && $this->isPreviousClientActionSuccessful();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doFetch(array $ids)
+    {
+        foreach ($this->client->getMulti($ids) as $id => $val) {
+            yield $id => $val;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doHave($id)
+    {
+        return $this->client->get($id) !== false
+            && $this->client->getResultCode() !== \Memcached::RES_NOTFOUND;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doDelete(array $ids)
+    {
+        $toDelete = count($ids);
+        foreach ((array) $this->client->deleteMulti($ids) as $result) {
```

Comment:
Older versions of the extension return a boolean stating "fully successful" or "one or more failures", so yes, IMHO.